pkg/controller/secretsyncer/secretsyncer.go

```diff
@@ -74,10 +78,35 @@ func AddToManager(mgr manager.Manager,
 			return requests
 		}),
 	); err != nil {
-		return fmt.Errorf("failed to create watch on reference cluster %s: %w", referenceClusterName, err)
+		return nil, fmt.Errorf("failed to create watch on reference cluster %s: %w", referenceClusterName, err)
 	}
 
-	return nil
+	enqueuer, src := newSource()
+
+	if err := c.Watch(src, handler.EnqueueRequestsFromMapFunc(func(o ctrlruntimeclient.Object) []reconcile.Request {
+		// The reference cluster enq all clusters
+		var requests []reconcile.Request
+		for _, targetCluster := range allTargetClusters.List() {
+			requests = append(requests, requestForCluster(targetCluster, o.GetNamespace(), o.GetName()))
+		}
+		return requests
+	})); err != nil {
+		return nil, fmt.Errorf("failed to create watch on reference cluster %s: %w", referenceClusterName, err)
+	}
+
+	return enqueuer, nil
+}
+
+func newSource() (Enqueuer, source.Source) {
```

Comment:
Please give this a more descriptive name, maybe something like `newConfigChangeSource`